src/providers/trusted_service/context/key_management.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+use super::ts_binding::{rpc_caller, rpc_caller_begin, rpc_caller_end, rpc_caller_invoke};
+use super::ts_protobuf::{
+    CloseKeyIn, DestroyKeyIn, DestroyKeyOut, GenerateKeyIn, GenerateKeyOut, KeyAttributes,
+    KeyLifetime, KeyPolicy, Opcode, OpenKeyIn, OpenKeyOut,
+};
+use super::Context;
+use log::{error, info};
+use parsec_interface::operations::psa_key_attributes::Attributes;
+use parsec_interface::requests::ResponseStatus;
+use prost::Message;
+use psa_crypto::types::status::{Error, Status};
+use std::convert::{TryFrom, TryInto};
+use std::mem::forget;
+use std::ptr::null_mut;
+
+impl Context {
+    pub fn generate_key(
+        &self,
+        key_attrs: Attributes,
+        id: u32,
+    ) -> parsec_interface::requests::Result<u32> {
+        let proto_req = Box::new(GenerateKeyIn {
+            attributes: Some(KeyAttributes {
+                r#type: u16::try_from(key_attrs.key_type)? as u32,
+                key_bits: key_attrs.bits.try_into()?,
+                lifetime: KeyLifetime::Persistent as u32,
+                id,
+                policy: Some(KeyPolicy {
+                    usage: key_attrs.policy.usage_flags.try_into()?,
+                    alg: key_attrs.policy.permitted_algorithms.try_into()?,
+                }),
+            }),
+        });
+        let mut proto_resp = Box::new(GenerateKeyOut::default());
+        Context::send_request(
+            &proto_req,
+            &mut proto_resp,
+            Opcode::GenerateKey,
+            self.rpc_caller,
+        )?;
+
+        Ok(0)
```

Comment:
What does the `u32` value returned mean?

Reply:
Good spot, I think that's a remnant from the first iterations

Reply:
Oh, sorry, thought I solved this, will update in a sec